background.js

```diff
@@ -1,39 +1,42 @@
-// Filter out utm_* query parameters
-function clean_utm(requestDetails) {
-    var url = new URL(requestDetails.url);
-    // console.debug("cleanURL utm_*", url.href);
-
-    if (url.search.length > 0) {
-        var params = url.searchParams;
-        var new_params = new URLSearchParams(params);
-        var needs_redirect = false;
-        for (let p of params.keys()) {
-            if (p.startsWith("utm_")) {
-                needs_redirect = true;
-                new_params.delete(p);
+function build_query_param_remover(shouldRemove) {
```

Comment:
If I understand correctly, `build_query_param_remover` can only remove all params.

Would it make sense to augment the parameters with a list of params to remove instead of a boolean?

Examples:
`['*']` = all params (old behaviour)
`['utm_*']` = for utm stuff
`['p1', 'p2']` = to manage URLs like `/path?p1=123&p2=456&necessary_param=dont_touch_me`

Reply:
This list can be (in a future release) set by user preferences

Reply:
Yeah that's probably better.

Reply:
yeah i'll get to this soon lol

Reply:
> If I understand correctly, build_query_param_remover can only remove all params.

It only removes a parameter `p` if `shouldRemove(p)` is true.

Reply:
Maybe we can do a mix of both.